src/mongodb/connection.js

```diff
@@ -0,0 +1,60 @@
+import mongoose from 'mongoose';
+import { EventEmitter } from 'events';
+import beautifyUniqueErrors from 'mongoose-beautiful-unique-validation';
+
+mongoose.Promise = Promise;
+mongoose.plugin(beautifyUniqueErrors);
+
+class MongodbConnection extends EventEmitter {
+  constructor(dbUrl) {
+    super();
+    this.name = 'mongodb';
+    this.dbUrl = dbUrl;
+  }
+
+  connect() {
+    const self = this;
```

Comment:
Because of how Javascript binds `this`, usually, you want to assign `this` to a variable to ensure the `this` you are referring to is what is intended. The callbacks provided to the event methods `on('open')` etc use arrow functions - we can assume `this` is unchanged inside those functions but better safe than sorry.

Reply:
I've actually removed it because in this context, the arrow functions will use this bound for this method